### PR TITLE
bug 1685406 - Add telemetry_mirror optional field to metrics schema

### PR DESCRIPTION
**Pull request by chutten**
### Pull Request checklist ###

- [ x] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [x ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x ] **Changelog**: This PR includes a changelog entry to `HISTORY.rst` or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

**Review by mdboom — CHANGES_REQUESTED** [review]
I think this also requires changes on the `Metrics.__init__` to store the value on the object, and then probably a small test update by adding this field to one of the example `metrics.yaml` files...

Also I assume (haven't checked the design doc) this is only applicable to certain metric types.  We should restrict that [using something like this](https://github.com/mozilla/glean_parser/blob/5d3db8e3fa3758ad24b93dc09aaef8dbd5571bbb/glean_parser/schemas/metrics.2-0-0.schema.yaml#L521) as well.

**Comment by chutten** [comment]
> I think this also requires changes on the `Metrics.__init__` to store the value on the object, and then probably a small test update by adding this field to one of the example `metrics.yaml` files...
> 
> Also I assume (haven't checked the design doc) this is only applicable to certain metric types. We should restrict that [using something like this](https://github.com/mozilla/glean_parser/blob/5d3db8e3fa3758ad24b93dc09aaef8dbd5571bbb/glean_parser/schemas/metrics.2-0-0.schema.yaml#L521) as well.

It applies to everything except Labeled String (and that's due to a gap in Telemetry's implementation) and JWE. I'm happy to enforce the types lining up properly in the FOG part of the build.

File: glean_parser/metrics.py [hunk 1]
```diff
@@ -90,6 +91,8 @@ def __init__(
                 getattr(DataSensitivity, x) for x in data_sensitivity
             ]
         self.defined_in = defined_in
+        if telemetry_mirror is not None:
```
Comment (mdboom) [inline glean_parser/metrics.py:94]: Should probably just remove the `if` statement so the member is always there, it just might be `None`.  That prevents downstream code from needing to write `if hasattr(metric, "telemetry_mirror")`...

**Review by mdboom — APPROVED** [review]
r+wc

File: glean_parser/metrics.py (hunk 1 above)
Comment (mdboom) [reply glean_parser/metrics.py:94]: As you rightly pointed out, this won't work (for schema validation reasons), so seems fine...

**Pull request merged**